api4/ldap.go

```diff
@@ -194,11 +197,35 @@ func linkLdapGroup(c *Context, w http.ResponseWriter, r *http.Request) {
 	} else {
 		// Group has never been linked
 		//
-		// TODO: In a future phase of LDAP groups sync `Name` will be used for at-mentions and will be editable on
-		// the front-end so it will not have an initial value of `model.NewId()` but rather a slugified version of
-		// the LDAP group name with an appended duplicate-breaker.
+		// `Name` will be used for at-mentions and will be editable on the front-end so
+		// create name based on display name and make sure it is unique.
+		// Append a duplicate-breaker if necessary.
+		// Only try a finite number of times, then give up.
+		baseName := strings.ReplaceAll(strings.ToLower(displayName), " ", "-")
+		validName := ""
+		for i := 0; i < 10; i++ {
```

Comment:
Why 10? The number itself is arbitrary. However, the calls made by ValidateGroupName do not distinguish between a db error and an item not found error. Therefore, if we don't find a valid name in 10 attempts, we will use the "NewId()" string. If the 10 was valid because of duplicate names, the admin would want to change the name anyway from "Group_10" to something more understandable. More than likely it is due to a db connection error.